manual/library/builtin.etex

```diff
@@ -190,7 +192,10 @@ exception Failure of string
 \index{Failure@\verb`Failure`}
 \begin{ocamldocdescription}
    Exception raised by library functions to signal that they are
-   undefined on the given arguments.
+   undefined on the given arguments.  The string gives some information
+   to the programmer; it is \emph{bad practice} to pattern match
```

Comment:
On Fri, 2 Oct 2015 09:59:34 -0700, Daniel Bünzli wrote:
> 
> What's wrong with pattern matching on result ? We have been using options for ages without having the option monad in the stdlib.

You of course can.  In many cases though, one cannot react immediately to the error but must propagate it.  Doing this by hand with repeated pattern matches is a pain.